### PR TITLE
feat(plasma-hope): maxWidth for Grid

### PR DESCRIPTION
**Pull request by kayman233**
Добавлена возможность настроить максимальную ширину компонента `Container`, входящего  в `Grid`, с помощью свойства `maxWidth`:
 
 ``` js
 <Container maxWidth="60rem">
 /* необходимые Row, Col */
</Container>
 ```

:baby_chick: Download canary assets:

[color_sberHealth_ios-swift--canary.495.4829781543.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_ios-swift--canary.495.4829781543.swift)  
[color_sberHealth_kotlin--canary.495.4829781543.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_kotlin--canary.495.4829781543.kt)  
[color_sberHealth_react-native--canary.495.4829781543.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberHealth_react-native--canary.495.4829781543.ts)  
[color_sbermarket_business_ios-swift--canary.495.4829781543.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_ios-swift--canary.495.4829781543.swift)  
[color_sbermarket_business_kotlin--canary.495.4829781543.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_kotlin--canary.495.4829781543.kt)  
[color_sbermarket_business_react-native--canary.495.4829781543.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_business_react-native--canary.495.4829781543.ts)  
[color_sbermarket_ios-swift--canary.495.4829781543.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_ios-swift--canary.495.4829781543.swift)  
[color_sbermarket_kotlin--canary.495.4829781543.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_kotlin--canary.495.4829781543.kt)  
[color_sbermarket_metro_ios-swift--canary.495.4829781543.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_ios-swift--canary.495.4829781543.swift)  
[color_sbermarket_metro_kotlin--canary.495.4829781543.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_kotlin--canary.495.4829781543.kt)  
[color_sbermarket_metro_react-native--canary.495.4829781543.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_metro_react-native--canary.495.4829781543.ts)  
[color_sbermarket_react-native--canary.495.4829781543.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_react-native--canary.495.4829781543.ts)  
[color_sbermarket_selgros_ios-swift--canary.495.4829781543.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_ios-swift--canary.495.4829781543.swift)  
[color_sbermarket_selgros_kotlin--canary.495.4829781543.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_kotlin--canary.495.4829781543.kt)  
[color_sbermarket_selgros_react-native--canary.495.4829781543.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_selgros_react-native--canary.495.4829781543.ts)  
[color_sbermarket_wlbusiness_ios-swift--canary.495.4829781543.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_ios-swift--canary.495.4829781543.swift)  
[color_sbermarket_wlbusiness_kotlin--canary.495.4829781543.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_kotlin--canary.495.4829781543.kt)  
[color_sbermarket_wlbusiness_react-native--canary.495.4829781543.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sbermarket_wlbusiness_react-native--canary.495.4829781543.ts)  
[color_sberprime_ios-swift--canary.495.4829781543.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_ios-swift--canary.495.4829781543.swift)  
[color_sberprime_kotlin--canary.495.4829781543.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_kotlin--canary.495.4829781543.kt)  
[color_sberprime_react-native--canary.495.4829781543.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/color_sberprime_react-native--canary.495.4829781543.ts)  
[shadow_sbermarket_react-native--canary.495.4829781543.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/shadow_sbermarket_react-native--canary.495.4829781543.ts)  
[typo_mage_ios-swift--canary.495.4829781543.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_ios-swift--canary.495.4829781543.swift)  
[typo_mage_kotlin--canary.495.4829781543.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_kotlin--canary.495.4829781543.kt)  
[typo_mage_react-native--canary.495.4829781543.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_mage_react-native--canary.495.4829781543.ts)  
[typo_plasma_ios-swift--canary.495.4829781543.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_ios-swift--canary.495.4829781543.swift)  
[typo_plasma_kotlin--canary.495.4829781543.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_kotlin--canary.495.4829781543.kt)  
[typo_plasma_react-native--canary.495.4829781543.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_plasma_react-native--canary.495.4829781543.ts)  
[typo_ruler_ios-swift--canary.495.4829781543.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_ios-swift--canary.495.4829781543.swift)  
[typo_ruler_kotlin--canary.495.4829781543.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_kotlin--canary.495.4829781543.kt)  
[typo_ruler_react-native--canary.495.4829781543.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_ruler_react-native--canary.495.4829781543.ts)  
[typo_sage_ios-swift--canary.495.4829781543.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_ios-swift--canary.495.4829781543.swift)  
[typo_sage_kotlin--canary.495.4829781543.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_kotlin--canary.495.4829781543.kt)  
[typo_sage_react-native--canary.495.4829781543.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sage_react-native--canary.495.4829781543.ts)  
[typo_sbermarket_ios-swift--canary.495.4829781543.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_ios-swift--canary.495.4829781543.swift)  
[typo_sbermarket_kotlin--canary.495.4829781543.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_kotlin--canary.495.4829781543.kt)  
[typo_sbermarket_react-native--canary.495.4829781543.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_sbermarket_react-native--canary.495.4829781543.ts)  
[typo_soulmate_ios-swift--canary.495.4829781543.swift](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_ios-swift--canary.495.4829781543.swift)  
[typo_soulmate_kotlin--canary.495.4829781543.kt](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_kotlin--canary.495.4829781543.kt)  
[typo_soulmate_react-native--canary.495.4829781543.ts](https://github.com/salute-developers/plasma/releases/download/v0.0.0-canary/typo_soulmate_react-native--canary.495.4829781543.ts)

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @salutejs/plasma-b2c@1.207.0-canary.495.4829781543.0
  npm install @salutejs/plasma-core@1.114.0-canary.495.4829781543.0
  npm install @salutejs/plasma-hope@1.207.0-canary.495.4829781543.0
  npm install @salutejs/plasma-icons@1.143.0-canary.495.4829781543.0
  npm install @salutejs/plasma-temple@1.158.0-canary.495.4829781543.0
  npm install @salutejs/plasma-tokens@1.49.0-canary.495.4829781543.0
  npm install @salutejs/plasma-ui@1.190.0-canary.495.4829781543.0
  npm install @salutejs/plasma-web@1.207.0-canary.495.4829781543.0
  npm install @salutejs/plasma-cy-utils@0.56.0-canary.495.4829781543.0
  npm install @salutejs/plasma-sb-utils@0.112.0-canary.495.4829781543.0
  # or 
  yarn add @salutejs/plasma-b2c@1.207.0-canary.495.4829781543.0
  yarn add @salutejs/plasma-core@1.114.0-canary.495.4829781543.0
  yarn add @salutejs/plasma-hope@1.207.0-canary.495.4829781543.0
  yarn add @salutejs/plasma-icons@1.143.0-canary.495.4829781543.0
  yarn add @salutejs/plasma-temple@1.158.0-canary.495.4829781543.0
  yarn add @salutejs/plasma-tokens@1.49.0-canary.495.4829781543.0
  yarn add @salutejs/plasma-ui@1.190.0-canary.495.4829781543.0
  yarn add @salutejs/plasma-web@1.207.0-canary.495.4829781543.0
  yarn add @salutejs/plasma-cy-utils@0.56.0-canary.495.4829781543.0
  yarn add @salutejs/plasma-sb-utils@0.112.0-canary.495.4829781543.0
  ```
</details>

### CODE REVIEW

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-495/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-495/

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-495/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-495/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-495/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-495/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-495/temple-storybook/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Theme Builder app deployed!
  
  http://plasma.sberdevices.ru/pr/plasma-theme-builder-pr-495/

**Comment by Salute-Eva** [comment]
Demo canvas app deployed!
  
  http://plasma.sberdevices.ru/pr/demo-canvas-app-pr-495/

**Comment by Salute-Eva** [comment]
**Component performance testing**

Result: **🟢 Success**

Check out report in job artifacts!

**Comment by Salute-Eva** [comment]
Documentation preview deployed!
  
  website: http://plasma.sberdevices.ru/pr/pr-495/
  ui storybook: http://plasma.sberdevices.ru/pr/pr-495/ui-storybook/
  web storybook: http://plasma.sberdevices.ru/pr/pr-495/web-storybook/
  b2c storybook: http://plasma.sberdevices.ru/pr/pr-495/b2c-storybook/
  temple storybook: http://plasma.sberdevices.ru/pr/pr-495/temple-storybook/

File: packages/plasma-hope/src/components/Grid/views/applyMaxWidth.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { css, InterpolationFunction } from 'styled-components';
+
+import { ContainerProps } from '../types';
+
+export const applyMaxWidth: InterpolationFunction<ContainerProps & { defaultMaxWidth?: string }> = ({
+    defaultMaxWidth = '125rem',
```
Comment (Yakutoc) [inline packages/plasma-hope/src/components/Grid/views/applyMaxWidth.tsx:6]: @kayman233 

Хочу уточнить, мы используем `defaultMaxWidth` потому что у нас 2 значения по-умолчанию, 125 и 90?

Я же правильно помню, что мы еще хотели Гене вопросы задать касательно этой задачи? Или уже все понятно?

Comment (kayman233) [reply packages/plasma-hope/src/components/Grid/views/applyMaxWidth.tsx:6]: по первому вопросу - да, чтобы не потерять обратную совместимость.

ну вообще по-хорошему стоит у Гены спросить. Надеюсь, завтра получится уточнить

**Review by Yakutoc — APPROVED** [review]

File: packages/plasma-hope/src/components/Grid/views/applyMaxWidth.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    maxWidth,
+}) =>
+    css`
+        max-width: ${maxWidth ?? defaultMaxWidth};
```
Comment (Yeti-or) [inline packages/plasma-hope/src/components/Grid/views/applyMaxWidth.tsx:10]: а вот эта штука у нас превращается во что-то при сборке? или остается как есть и только в современных бро работает?

File: packages/plasma-hope/src/components/Grid/views/applyMaxWidth.tsx (hunk 1 above)
Comment (Yeti-or) [reply packages/plasma-hope/src/components/Grid/views/applyMaxWidth.tsx:6]: надо задачу по крайней мере завести на то чтобы выровнять эти штуки

**Review by Yeti-or — APPROVED** [review]

File: packages/plasma-hope/src/components/Grid/views/applyMaxWidth.tsx (hunk 2 above)
Comment (kayman233) [reply packages/plasma-hope/src/components/Grid/views/applyMaxWidth.tsx:10]: да, как я посмотрел, там при сборке сразу они подставляются в оба варианта контейнера b2c/web. Так что должно работать

**Pull request merged**